bindings/python/triplestore/triplestore.py

```diff
@@ -0,0 +1,608 @@
+
+    def __getitem__(self, key):
+        return self.uri + key
+
+    def __repr__(self):
+        return f"Namespace({self.iri})"
+
+    def __str__(self):
+        return self.uri
+
+
+# Pre-defined namespaces
+XML = Namespace("http://www.w3.org/XML/1998/namespace")
+RDF = Namespace("http://www.w3.org/1999/02/22-rdf-syntax-ns#")
+RDFS = Namespace("http://www.w3.org/2000/01/rdf-schema#")
+XSD = Namespace("http://www.w3.org/2001/XMLSchema#")
+OWL = Namespace("http://www.w3.org/2002/07/owl#")
+SKOS = Namespace("http://www.w3.org/2004/02/skos/core#")
+DC = Namespace("http://purl.org/dc/elements/1.1/")
+DCTERMS = Namespace("http://purl.org/dc/terms/")
+FOAF = Namespace("http://xmlns.com/foaf/0.1/")
+DOAP = Namespace("http://usefulinc.com/ns/doap#")
+FNO = Namespace("https://w3id.org/function/ontology#")
+
+EMMO = Namespace("http://emmo.info/emmo#")
+MAP = Namespace("http://emmo.info/domain-mappings#")
+DM = Namespace("http://emmo.info/datamodel#")
+
+class Literal(str):
+    """A literal RDF value."""
+    def __new__(cls, value, lang=None, datatype=None):
+        string = super().__new__(cls, value)
+        if lang:
+            if datatype:
+                raise TypeError("A literal can only have one of `lang` or "
+                                "`datatype`.")
+            string.lang = str(lang)
+            string.datatype = None
+        else:
+            if datatype:
+                d = {
+                    str: XSD.string,
+                    bool: XSD.boolean,
+                    int: XSD.integer,
+                    float: XSD.double,
+                    bytes: XSD.hexBinary,
+                    bytearray: XSD.hexBinary,
+                    datetime: XSD.dateTime,
+                }
+                string.datatype = d.get(datatype, datatype)
+            elif isinstance(value, str):
+                string.datatype = None
+            elif isinstance(value, bool):
+                string.datatype = XSD.boolean
+            elif isinstance(value, int):
+                string.datatype = XSD.integer
+            elif isinstance(value, float):
+                string.datatype = XSD.double
+            elif isinstance(value, (bytes, bytearray)):
+                string = value.hex()
+                string.datatype = XSD.hexBinary
+            elif isinstance(value, datetime):
+                string.datatype = XSD.dateTime
+            # TODO:
+            #   - XSD.base64Binary
+            #   - XSD.byte, XSD.unsignedByte
+            else:
+                string.datatype = None
+            string.lang = None
+        return string
+
+    def __repr__(self):
+        lang = f", lang='{self.lang}'" if self.lang else ""
+        datatype = f", datatype='{self.datatype}'" if self.datatype else ""
+        return f"Literal('{self}'{lang}{datatype})"
+
+    value = property(
+        fget=lambda self: self.toPython(),
+        doc="Appropriate python datatype derived from this RDF literal.",
+    )
+
+    def toPython(self):
```

Comment:
CamelCase is typically reserved for classes, this should probably be to_python

Reply:
Good point. The reason for the CamelCase was to be consistent with rdflib. But I agree with you that internal consistency is more important than consistency with rdflib.

The function is now renamed to `to_python()`.